src/components/transport-mode/transport-icon/index.tsx

```diff
@@ -42,6 +42,34 @@ export function TransportIcon({ mode }: TransportIconProps) {
   );
 }
 
+export type TransportIconWithLabelProps = {
+  mode: TransportModeGroup;
+  label?: string | null;
+};
+
+export function TransportIconWithLabel({
+  mode,
+  label,
+}: TransportIconWithLabelProps) {
+  const { t } = useTranslation();
+  const { backgroundColor, overrideMode } = useTransportationThemeColor(mode);
+  return (
+    <span className={style.transportIconWithLabel} style={{ backgroundColor }}>
+      <MonoIcon
+        icon={getTransportModeIcon(mode)}
+        role="img"
+        alt={t(transportModeToTranslatedString(mode))}
+        overrideMode={overrideMode}
+      />
+      {label && (
+        <span style={{ color: overrideMode === 'light' ? 'black' : 'white' }}>
```

Comment:
I think we should use the `text-colors-primary` color for the label here 👍

Reply:
That won't work. The label needs to be the same color as the icon and can't change depending on the set theme. 

I'll add this to the todo, to figure out how to solve it better.